docs/tutorials/developer-journey/level-3/3.2-https-outcalls.md

```diff
@@ -0,0 +1,813 @@
+    let host : Text = "api.pro.coinbase.com";
+    let url = "https://" # host # "/products/ICP-USD/candles?start=" # Nat64.toText(start_timestamp) # "&end=" # Nat64.toText(start_timestamp) # "&granularity=" # Nat64.toText(ONE_MINUTE);
+
+    // Then prepare headers for the system http_request call
+
+    let request_headers = [
+        { name = "Host"; value = host # ":443" },
+        { name = "User-Agent"; value = "exchange_rate_canister" },
+    ];
+
+    // Then transform the context
+
+    let transform_context : Types.TransformContext = {
+      function = transform;
+      context = Blob.fromArray([]);
+    };
+
+    // Finally, define the HTTP request
+
+    let http_request : Types.HttpRequestArgs = {
+        url = url;
+        max_response_bytes = null; //optional for request
+        headers = request_headers;
+        body = null; //optional for request
+        method = #get;
+        transform = ?transform_context;
+    };
+
+    // Now, we need to add some cycles to our call, since cycles to pay for the call must be transferred with the call. 
+    // The way Cycles.add() works is that it adds those cycles to the next asynchronous call.
+    // "Function add(amount) indicates the additional amount of cycles to be transferred in the next remote call".
+    // See: https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-http_request
+
+    Cycles.add(20_949_972_000);
+    
+    // Now that we have the HTTP request and cycles to send with the call, we can make the HTTP request and await the response.
+
+    let http_response : Types.HttpResponsePayload = await ic.http_request(http_request);
+    
+    // Once we have the response, we need to decode it. The body of the HTTP response should come back as [Nat8], which needs to be decoded into readable text. 
+    // To do this, we:
+    //  1. Convert the [Nat8] into a Blob
+    //  2. Use Blob.decodeUtf8() method to convert the Blob to a ?Text optional 
+    //  3. We use a switch to explicitly call out both cases of decoding the Blob into ?Text
+
+    let response_body: Blob = Blob.fromArray(http_response.body);
+    let decoded_text: Text = switch (Text.decodeUtf8(response_body)) {
+        case (null) { "No value returned" };
+        case (?y) { y };
+    };
+
+    // Finally, we can return the response of the body.
+
+    // The API response will looks like this:
+
+    // ("[[1682978460,5.714,5.718,5.714,5.714,243.5678]]")
+
+    // Which can be formatted as this
+    //  [
+    //     [
+    //         1682978460, <-- start/timestamp
+    //         5.714, <-- low
+    //         5.718, <-- high
+    //         5.714, <-- open
+    //         5.714, <-- close
+    //         243.5678 <-- volume
+    //     ],
+    // ]
+
+    decoded_text
+  };
+
+    // As a final step, create a function that transform's the raw content into an HTTP payload:
+    public query func transform(raw : Types.TransformArgs) : async Types.CanisterHttpResponsePayload {
```

Comment:
it looks like the `transform` method is defined twice. is that intentional?

Reply:
if so, some explanation as to why would be good. I got hung up on this while reading through the guide.